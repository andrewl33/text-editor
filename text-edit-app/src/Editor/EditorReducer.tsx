import {
  ADD_TAG_FAILURE,
  ADD_TAG_REQUEST,
  ADD_TAG_SUCCESS,
  ADD_USER_TO_FILE_FAILURE,
  ADD_USER_TO_FILE_REQUEST,
  ADD_USER_TO_FILE_SUCCESS,
  AUTH_FILE_FAILURE,
  AUTH_FILE_REQUEST,
  AUTH_FILE_SUCCESS,
  CHANGE_FILE_NAME_FAILURE,
  CHANGE_FILE_NAME_REQUEST,
  CHANGE_FILE_NAME_SUCCESS,
  CHANGED_CODE,
  CLOSE_ALERT,
  GET_TEXT_AUTH,
  GET_TEXT_FAILURE,
  GET_TEXT_REQUEST,
  GET_TEXT_SUCCESS,
  LOCAL_UPDATE_CODE,
  LOCK_TEXT_FAILURE,
  LOCK_TEXT_REQUEST,
  LOCK_TEXT_SUCCESS,
  NEW_TEXT_FAILURE,
  NEW_TEXT_REQUEST,
  NEW_TEXT_SUCCESS,
  REMOVE_TAG_FAILURE,
  REMOVE_TAG_REQUEST,
  REMOVE_TAG_SUCCESS,
  REMOVE_USER_FROM_FILE_FAILURE,
  REMOVE_USER_FROM_FILE_REQUEST,
  REMOVE_USER_FROM_FILE_SUCCESS,
  SHARE_LINK,
  UPDATE_CODE_FAILURE,
  UPDATE_CODE_REQUEST,
  UPDATE_CODE_SUCCESS
} from "../constants";
import { EditorStoreState } from "../types";
import { EditorAction } from "./EditorAction";

export const initialState: EditorStoreState = {
  codeText: "",
  tags: [],
  isNewPage: false,
  isLocked: false,
  isSaved: true,
  openAlert: false,
  alertMessage: "",
  filePrompt: false,
  createDate: "",
  users: [],
  name: "",
  remountEditorComponent: 0
};

export const editorReducer = (
  state: EditorStoreState = initialState,
  action: EditorAction
) => {
  switch (action.type) {
    case LOCAL_UPDATE_CODE:
      return { ...state, codeText: action.payload.codeText };

    case UPDATE_CODE_REQUEST:
      return { ...state };
    case UPDATE_CODE_SUCCESS:
      return {
        ...state,
        isSaved: true,
        openAlert: true,
        alertMessage: "Updated code"
      };
    case UPDATE_CODE_FAILURE:
      return {
        ...state,

        isSaved: false,
        openAlert: true,
        alertMessage: "Failed to update code"
      };

    case GET_TEXT_REQUEST:
      return {
        ...state,
        openAlert: true,
        alertMessage: "Loading code..."
      };
    case GET_TEXT_SUCCESS:
      if (action.payload && action.payload.body) {
        return {
          ...initialState,
          codeText: action.payload.body.codeText,
          tags: action.payload.body.tags,
          name: action.payload.body.name,
          isLocked: action.payload.body.isLocked,
          isSaved: true,
          users: action.payload.body.users,
          createDate: action.payload.body.createDate,
          remountEditorComponent: state.remountEditorComponent + 1
        };
      }

    case GET_TEXT_FAILURE:
      return {
        ...state,

        isSaved: false,
        openAlert: true,
        alertMessage: "Failed to load saved code"
      };
    case GET_TEXT_AUTH:
      return { ...state, filePrompt: true, prompt: "Private File" };

    case NEW_TEXT_REQUEST:
      return { ...state, isNewPage: true };
    case NEW_TEXT_SUCCESS:
      return {
        ...initialState,
        isNewPage: true,
        openAlert: true,
        alertMessage: "Created a new page"
      };
    case NEW_TEXT_FAILURE:
      return {
        ...state,
        isNewPage: false,
        openAlert: true,
        alertMessage: "Failed to create new page, try again"
      };

    case AUTH_FILE_REQUEST:
      return { ...state, openAlert: true, alertMessage: "Authenticating" };
    case AUTH_FILE_SUCCESS:
      return { ...state, filePrompt: false };
    case AUTH_FILE_FAILURE:
      return {
        ...state,
        openAlert: true,
        alertMessage: "Wrong Password"
      };

    case CHANGED_CODE:
      return state;

    case ADD_TAG_REQUEST:
      return state;
    case ADD_TAG_SUCCESS:
      return {
        ...state,
        tags: [...state.tags, action.payload && action.payload.name]
      };
    case ADD_TAG_FAILURE:
      return {
        ...state,
        openAlert: true,
        alertMessage: "Could not connect to DB!"
      };

    case REMOVE_TAG_REQUEST:
      return state;
    case REMOVE_TAG_SUCCESS:
      const name = action.payload && action.payload.name && action.payload.name;
      return {
        ...state,
        tags: state.tags.filter(t => t !== name)
      };
    case REMOVE_TAG_FAILURE:
      return {
        ...state,
        openAlert: true,
        alertMessage: "Could not connect to DB"
      };

    case CHANGE_FILE_NAME_REQUEST:
      return state;
    case CHANGE_FILE_NAME_SUCCESS:
      return { ...state, name: action.payload && action.payload.name };
    case CHANGE_FILE_NAME_FAILURE:
      return { ...state, alertMessage: "Could not connect to DB" };

    case ADD_USER_TO_FILE_FAILURE:
      return state;
    case ADD_USER_TO_FILE_SUCCESS:
      return {
        ...state,
        users: [...state.users, action.payload && action.payload.accountName]
      };
    case ADD_USER_TO_FILE_REQUEST:
      return state;

    case REMOVE_USER_FROM_FILE_FAILURE:
      return state;
    case REMOVE_USER_FROM_FILE_SUCCESS:
      if (action.payload && action.payload.accountName) {
        return {
          ...state,
          users: state.users.filter(u => u !== action.payload!.accountName)
        };
      } else {
        return state;
      }
    case REMOVE_USER_FROM_FILE_REQUEST:
      return { ...state, alertMessage: "Could not connect to DB" };

    case LOCK_TEXT_REQUEST:
      return state;
    case LOCK_TEXT_SUCCESS:
      return { ...state, isLocked: true };
    case LOCK_TEXT_FAILURE:
      return state;

    case SHARE_LINK:
      return { ...state, openAlert: true, alertMessage: "Copied to clipboard" };
    case CLOSE_ALERT:
      return { ...state, openAlert: false };
    default:
      return state;
  }
};
