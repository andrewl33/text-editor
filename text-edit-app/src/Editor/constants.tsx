// editor
export const LOCAL_UPDATE_CODE = "LOCAL_UPDATE_CODE";
export type LOCAL_UPDATE_CODE = typeof LOCAL_UPDATE_CODE;

export const UPDATE_CODE_REQUEST = "UPDATE_CODE_REQUEST";
export type UPDATE_CODE_REQUEST = typeof UPDATE_CODE_REQUEST;

export const UPDATE_CODE_SUCCESS = "UPDATE_CODE_SUCCESS";
export type UPDATE_CODE_SUCCESS = typeof UPDATE_CODE_SUCCESS;

export const UPDATE_CODE_FAILURE = "UPDATE_CODE_FAILURE";
export type UPDATE_CODE_FAILURE = typeof UPDATE_CODE_FAILURE;

export const CHANGED_CODE = "CHANGED_CODE";
export type CHANGED_CODE = typeof CHANGED_CODE;

export const NEW_TEXT_REQUEST = "NEW_TEXT_REQUEST";
export type NEW_TEXT_REQUEST = typeof NEW_TEXT_REQUEST;

export const NEW_TEXT_SUCCESS = "NEW_TEXT_SUCCESS";
export type NEW_TEXT_SUCCESS = typeof NEW_TEXT_SUCCESS;

export const NEW_TEXT_FAILURE = "NEW_TEXT_FAILURE";
export type NEW_TEXT_FAILURE = typeof NEW_TEXT_FAILURE;

export const GET_TEXT_REQUEST = "GET_TEXT_REQUEST";
export type GET_TEXT_REQUEST = typeof GET_TEXT_REQUEST;

export const GET_TEXT_SUCCESS = "GET_TEXT_SUCCESS";
export type GET_TEXT_SUCCESS = typeof GET_TEXT_SUCCESS;

export const GET_TEXT_FAILURE = "GET_TEXT_FAILURE";
export type GET_TEXT_FAILURE = typeof GET_TEXT_FAILURE;

export const GET_TEXT_AUTH = "GET_TEXT_AUTH";
export type GET_TEXT_AUTH = typeof GET_TEXT_AUTH;

export const AUTH_FILE_REQUEST = "AUTH_FILE_REQUEST";
export type AUTH_FILE_REQUEST = typeof AUTH_FILE_REQUEST;

export const AUTH_FILE_SUCCESS = "AUTH_FILE_SUCCESS";
export type AUTH_FILE_SUCCESS = typeof AUTH_FILE_SUCCESS;

export const AUTH_FILE_FAILURE = "AUTH_FILE_FAILURE";
export type AUTH_FILE_FAILURE = typeof AUTH_FILE_FAILURE;

export const ADD_TAG_REQUEST = "ADD_TAG_REQUEST";
export type ADD_TAG_REQUEST = typeof ADD_TAG_REQUEST;

export const ADD_TAG_SUCCESS = "ADD_TAG_SUCCESS";
export type ADD_TAG_SUCCESS = typeof ADD_TAG_SUCCESS;

export const ADD_TAG_FAILURE = "ADD_TAG_FAILURE";
export type ADD_TAG_FAILURE = typeof ADD_TAG_FAILURE;

export const REMOVE_TAG_REQUEST = "REMOVE_TAG_REQUEST";
export type REMOVE_TAG_REQUEST = typeof REMOVE_TAG_REQUEST;

export const REMOVE_TAG_SUCCESS = "REMOVE_TAG_SUCCESS";
export type REMOVE_TAG_SUCCESS = typeof REMOVE_TAG_SUCCESS;

export const REMOVE_TAG_FAILURE = "REMOVE_TAG_FAILURE";
export type REMOVE_TAG_FAILURE = typeof REMOVE_TAG_FAILURE;

export const CHANGE_FILE_NAME_REQUEST = "CHANGE_FILE_NAME_REQUEST";
export type CHANGE_FILE_NAME_REQUEST = typeof CHANGE_FILE_NAME_REQUEST;

export const CHANGE_FILE_NAME_SUCCESS = "CHANGE_FILE_NAME_SUCCESS";
export type CHANGE_FILE_NAME_SUCCESS = typeof CHANGE_FILE_NAME_SUCCESS;

export const CHANGE_FILE_NAME_FAILURE = "CHANGE_FILE_NAME_FAILURE";
export type CHANGE_FILE_NAME_FAILURE = typeof CHANGE_FILE_NAME_FAILURE;

export const LOCK_TEXT_REQUEST = "LOCK_TEXT_REQUEST";
export type LOCK_TEXT_REQUEST = typeof LOCK_TEXT_REQUEST;

export const ADD_USER_TO_FILE_REQUEST = "ADD_USER_TO_FILE_REQUEST";
export type ADD_USER_TO_FILE_REQUEST = typeof ADD_USER_TO_FILE_REQUEST;

export const ADD_USER_TO_FILE_SUCCESS = "ADD_USER_TO_FILE_SUCCESS";
export type ADD_USER_TO_FILE_SUCCESS = typeof ADD_USER_TO_FILE_SUCCESS;

export const ADD_USER_TO_FILE_FAILURE = "ADD_USER_TO_FILE_FAILURE";
export type ADD_USER_TO_FILE_FAILURE = typeof ADD_USER_TO_FILE_FAILURE;

export const REMOVE_USER_FROM_FILE_REQUEST = "REMOVE_USER_FROM_FILE_REQUEST";
export type REMOVE_USER_FROM_FILE_REQUEST = typeof REMOVE_USER_FROM_FILE_REQUEST;

export const REMOVE_USER_FROM_FILE_SUCCESS = "REMOVE_USER_FROM_FILE_SUCCESS";
export type REMOVE_USER_FROM_FILE_SUCCESS = typeof REMOVE_USER_FROM_FILE_SUCCESS;

export const REMOVE_USER_FROM_FILE_FAILURE = "REMOVE_USER_FROM_FILE_FAILURE";
export type REMOVE_USER_FROM_FILE_FAILURE = typeof REMOVE_USER_FROM_FILE_FAILURE;

export const LOCK_TEXT_SUCCESS = "LOCK_TEXT_SUCCESS";
export type LOCK_TEXT_SUCCESS = typeof LOCK_TEXT_SUCCESS;

export const LOCK_TEXT_FAILURE = "LOCK_TEXT_FAILURE";
export type LOCK_TEXT_FAILURE = typeof LOCK_TEXT_FAILURE;
