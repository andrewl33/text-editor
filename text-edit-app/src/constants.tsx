export const NEW_TEXT = 'NEW_TEXT';
export type NEW_TEXT = typeof NEW_TEXT;

export const LOCK_TEXT = 'LOCK_TEXT';
export type LOCK_TEXT = typeof LOCK_TEXT;

export const SHARE_LINK = 'SHARE_LINK';
export type SHARE_LINK = typeof SHARE_LINK;