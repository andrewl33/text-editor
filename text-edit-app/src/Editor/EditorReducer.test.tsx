import * as constants from "../constants";
import { editorReducer, initialState } from "./EditorReducer";

describe("EditorReducer", () => {
  test("it should return initial state", () => {
    expect(editorReducer(undefined, {} as any)).toMatchSnapshot();
  });

  describe("updateCode", () => {
    test("it should handle UPDATE_CODE_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.UPDATE_CODE_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle UPDATE_CODE_SUCCESS", () => {
      expect(
        editorReducer(undefined, { type: constants.UPDATE_CODE_SUCCESS })
      ).toMatchSnapshot();
    });

    test("it should handle UPDATE_CODE_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.UPDATE_CODE_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("get text", () => {
    test("it should handle GET_TEXT_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.GET_TEXT_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle GET_TEXT_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.GET_TEXT_SUCCESS,
          payload: {
            success: true,
            body: {
              codeText: "test string",
              tags: [],
              name: "",
              createDate: "1-1-2001",
              isLocked: false,
              users: []
            }
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle GET_TEXT_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.GET_TEXT_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("lock text", () => {
    test("it should handle LOCK_TEXT_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.LOCK_TEXT_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle LOCK_TEXT_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.LOCK_TEXT_SUCCESS,
          payload: {
            success: true,
            message: "Authenticated"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle LOCK_TEXT_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.LOCK_TEXT_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("add tag", () => {
    test("it should handle ADD_TAG_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.ADD_TAG_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_TAG_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.ADD_TAG_SUCCESS,
          payload: {
            success: true,
            name: "newTag"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_TAG_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.ADD_TAG_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("remove tag", () => {
    test("it should handle REMOVE_TAG_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.REMOVE_TAG_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_TAG_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.REMOVE_TAG_SUCCESS,
          payload: {
            success: true,
            name: "tag"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_TAG_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.REMOVE_TAG_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("new text", () => {
    const request = { ...initialState, isNewPage: true };
    const success = {
      ...initialState,
      isNewPage: true,
      openAlert: true,
      alertMessage: "Created a new page"
    };
    const failure = {
      ...initialState,
      isNewPage: false,
      openAlert: true,
      alertMessage: "Failed to create new page, try again"
    };

    test("it should handle NEW_TEXT_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.NEW_TEXT_REQUEST })
      ).toEqual(request);
    });

    test("it should handle NEW_TEXT_SUCCESS", () => {
      expect(
        editorReducer(undefined, { type: constants.NEW_TEXT_SUCCESS })
      ).toEqual(success);
    });

    test("it should handle NEW_TEXT_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.NEW_TEXT_FAILURE })
      ).toEqual(failure);
    });
  });

  describe("auth file", () => {
    test("it should handle AUTH_FILE_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.AUTH_FILE_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle AUTH_FILE_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.AUTH_FILE_SUCCESS,
          payload: {
            success: true,
            message: "Authenticated"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle AUTH_FILE_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.AUTH_FILE_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("change file name", () => {
    test("it should handle CHANGE_FILE_NAME_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.CHANGE_FILE_NAME_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle CHANGE_FILE_NAME_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.CHANGE_FILE_NAME_SUCCESS,
          payload: {
            success: true,
            name: "newName"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle CHANGE_FILE_NAME_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.CHANGE_FILE_NAME_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("add user to file", () => {
    test("it should handle ADD_USER_TO_FILE_REQUEST", () => {
      expect(
        editorReducer(undefined, { type: constants.ADD_USER_TO_FILE_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_USER_TO_FILE_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.ADD_USER_TO_FILE_SUCCESS,
          payload: {
            success: true,
            accountName: "newName"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_USER_TO_FILE_FAILURE", () => {
      expect(
        editorReducer(undefined, { type: constants.ADD_USER_TO_FILE_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("remove user from file", () => {
    test("it should handle REMOVE_USER_FROM_FILE_REQUEST", () => {
      expect(
        editorReducer(undefined, {
          type: constants.REMOVE_USER_FROM_FILE_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_USER_FROM_FILE_SUCCESS", () => {
      expect(
        editorReducer(undefined, {
          type: constants.REMOVE_USER_FROM_FILE_SUCCESS,
          payload: {
            success: true,
            accountName: "newName"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_USER_FROM_FILE_FAILURE", () => {
      expect(
        editorReducer(undefined, {
          type: constants.REMOVE_USER_FROM_FILE_FAILURE
        })
      ).toMatchSnapshot();
    });
  });

  // type changed code, lock text, and share link
  describe("EditorReducer: local actions", () => {
    test("it should handle CLOSE_ALERT", () => {
      expect(editorReducer(undefined, { type: constants.CLOSE_ALERT })).toEqual(
        {
          ...initialState,
          openAlert: false
        }
      );
    });
  });
});
