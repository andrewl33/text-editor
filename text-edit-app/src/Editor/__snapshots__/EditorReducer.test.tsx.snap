// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorReducer add tag it should handle ADD_TAG_FAILURE 1`] = `
Object {
  "alertMessage": "Could not connect to DB!",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer add tag it should handle ADD_TAG_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer add tag it should handle ADD_TAG_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [
    "newTag",
  ],
  "users": Array [],
}
`;

exports[`EditorReducer add user to file it should handle ADD_USER_TO_FILE_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer add user to file it should handle ADD_USER_TO_FILE_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer add user to file it should handle ADD_USER_TO_FILE_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [
    "newName",
  ],
}
`;

exports[`EditorReducer auth file it should handle AUTH_FILE_FAILURE 1`] = `
Object {
  "alertMessage": "Wrong Password",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer auth file it should handle AUTH_FILE_REQUEST 1`] = `
Object {
  "alertMessage": "Authenticating",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer auth file it should handle AUTH_FILE_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer change file name it should handle CHANGE_FILE_NAME_FAILURE 1`] = `
Object {
  "alertMessage": "Could not connect to DB",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer change file name it should handle CHANGE_FILE_NAME_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer change file name it should handle CHANGE_FILE_NAME_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "newName",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer get text it should handle GET_TEXT_FAILURE 1`] = `
Object {
  "alertMessage": "Failed to load saved code",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": false,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer get text it should handle GET_TEXT_REQUEST 1`] = `
Object {
  "alertMessage": "Loading code...",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer get text it should handle GET_TEXT_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "test string",
  "createDate": "1-1-2001",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 1,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer it should return initial state 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer lock text it should handle LOCK_TEXT_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer lock text it should handle LOCK_TEXT_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer lock text it should handle LOCK_TEXT_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": true,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer new text it should handle NEW_TEXT_FAILURE 1`] = `
Object {
  "alertMessage": "Failed to create new page, try again",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer new text it should handle NEW_TEXT_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": true,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer new text it should handle NEW_TEXT_SUCCESS 1`] = `
Object {
  "alertMessage": "Created a new page",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": true,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer remove tag it should handle REMOVE_TAG_FAILURE 1`] = `
Object {
  "alertMessage": "Could not connect to DB",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer remove tag it should handle REMOVE_TAG_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer remove tag it should handle REMOVE_TAG_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer remove user from file it should handle REMOVE_USER_FROM_FILE_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer remove user from file it should handle REMOVE_USER_FROM_FILE_REQUEST 1`] = `
Object {
  "alertMessage": "Could not connect to DB",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer remove user from file it should handle REMOVE_USER_FROM_FILE_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer updateCode it should handle UPDATE_CODE_FAILURE 1`] = `
Object {
  "alertMessage": "Failed to update code",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": false,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer updateCode it should handle UPDATE_CODE_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": false,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;

exports[`EditorReducer updateCode it should handle UPDATE_CODE_SUCCESS 1`] = `
Object {
  "alertMessage": "Updated code",
  "codeText": "",
  "createDate": "",
  "filePrompt": false,
  "isLocked": false,
  "isNewPage": false,
  "isSaved": true,
  "name": "",
  "openAlert": true,
  "remountEditorComponent": 0,
  "tags": Array [],
  "users": Array [],
}
`;
